dnsapi/dns_synology_dsm.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# Here is a script to add/remove TXT records to DNS Server on Synology DSM
+#
+# Author: Arabezar aka Arkadii Zhuchenko © 13.07.2020
+# Great thanks to loderunner84 for the invaluable help in synowebapi research
+#
+#returns 0 means success, otherwise error.
+
+_DNS_TTL="1"
+
+########  Public functions #####################
+
+dns_synology_dsm_add() {
+
+  _info "Using API for Synology DSM - adding TXT to Synology DNS Server"
+  
+  fulldomain=$1
+  txtvalue=$2
+  _debug fulldomain "$fulldomain"
+  _debug txtvalue "$txtvalue"
+
+  maindomain="${fulldomain//_acme-challenge\./}"
+  _debug3 maindomain "$maindomain"
+
+  # SynoWebAPI call can be replaced by adding the line to the "/var/packages/DNSServer/target/named/etc/zone/master/$maindomain" file
+  response=$(synowebapi --exec api=SYNO.DNSServer.Zone.Record method=create version=1 runner=admin \
```

Comment:
You're right, this script is intended to be run on Synology only because it is for Synology DNS Server only and it has to automate the process of getting cert with no human participation for all Synology applications that need SSL-cert ([synology_dsm deploy hook](https://github.com/acmesh-official/acme.sh/blob/master/deploy/synology_dsm.sh) is also a part of the whole process) using Synology Task Manager.
Calls via HTTP/HTTPS are not recommended because the script should store admin user and password. I would also like to rewrite the `synology_dsm.sh` code but still have problems deploying cert through `synowebapi`. I hope that will happen somewhere in time.

I've added the Requirements to the wiki.

Reply:
@tresni There is a big community of Synology home users. As usual, we have our infrastructure just on a single Syno device, and we're going to run this script just on Synology.